Issue #365 - Added option to override package manager.

RobinTo opened the pull request:
Implemented parameter for create-razzle-app to override package manager, could resolve #365 if useable.

`create-razzle-app my-project -p npm`

Command|Info|Valid values
--- | --- | ---
`-p --packageManager`|Override package manager, default to yarn.|`npm`, `yarn`

Tested this on mac using
* `npx ./razzle/packages/create-razzle-app/bin/create-razzle-app test-razzle-project -p npm`
* `cd test-razzle-project`
* `npm start`

If the option is not supplied it defaults to old behaviour, otherwise checks that it is one of `yarn` or `npm`, and defaults to old behaviour if not. Could remove check for `yarn` or `npm` if you want to enable any value to be used.

> I want to be able to keep supporting projects that use yarn, while favoring npm for new projects.

Due to the sentence above I was unsure wether to set the default to npm, but currently it default to yarn as previously done. Could be changed if desired.

I hope you find this useful, even though the issue was marked as low priority. 

Some yarn lockfiles unrelated to my modifications appear to have many changes. I'm not very familiar with the inner workings of yarn, but I suspect it might be due to newer yarn version when I ran the initial yarn setup?

Commit:
Issue#365 - Added option to override package manager.

Commit:
Added missign letter in readme.

jariz commented:
I'm slowly seeing the JS hemisphere moving back to npm, but correct me if I'm wrong.
Eitherway, yarn was the previous default so probably best to keep it that way.

jariz commented:
@RobinTo Can you update all readme's? There are more than 2.
I know this is kind of annoying but it's something that's being worked on (#409)

Commit:
Updated the last readme.

Commit:
Modified cli options in readme toc.

RobinTo commented:
Of course, all the relevant readme files should now be updated.

jaredpalmer commented on packages/create-razzle-app/lib/messages.js:
```diff
@@ -22,6 +22,10 @@ exports.exampleHelp = function() {
 };
 
+exports.packageManagerHelp = function() {
+  return `Wich package manager (npm or yarn) to utilize for setup.`;
```
Which***

jaredpalmer reviewed (commented):
Not sure why lockfiles are in this commit. May need to open a new PR

jaredpalmer commented:
Is there a way to automatically tell based on whether user has called this with `npx` or `yarn create`?

Commit:
Fixed spelling in cli help message.

RobinTo commented:
I fixed the spelling error. Let me know if you want me to open a new pull request where I reset the lockfiles to the versions present on current master.

As far as I could see with some light searching, there is no way to detect if it was done through yarn create or directly, since according to https://yarnpkg.com/lang/en/docs/cli/create/ yarn create is equal to

```
$ yarn global add create-react-app
$ create-react-app my-app
```

where the end result is the same as if you had installed create-react-app globally with any packagemanager and ran the command directly. If you have any suggestions where to look for making it work, please let me know though and I'd be happy to look into it further.

jariz commented:
I've done a little research by running `yarn create` and sadly [`process.argv`](https://nodejs.org/api/process.html#process_process_argv) returns the same as it would when directly running the script.
On top of that [`process.ppid`](https://nodejs.org/api/process.html#process_process_ppid) is set to undefined, so no way to get information about who started the script either.

So, short answer to @jaredpalmer's question: Nope.

jariz commented:
Scratch that, ppid only exists in 9.2+ and it does appear to return the yarn pid, but then again, that version has only been out for 2 months.
So, I think this is kind of out of scope for now, unless we're willing to do all sorts of [crazy platform specific hacks](https://stackoverflow.com/questions/5566009/node-js-parent-process-id).

jaredpalmer committed:
Bump deps

jaredpalmer committed:
v0.8.13

jaredpalmer committed:
Cosmetic change

jaredpalmer committed:
v0.8.14

fix-fix committed:
Fix chunk links in with-react-loadable example (#550)

Fix incorrect link to chunk scripts caused by string concatenation instead of incrementing the port number value.

`process.env.*` values are strings, so in the example file port value was `'30001'` instead of `3001`.

marklearst committed:
Patching for IE11 (#547)

Mirroring the patch for react-dev-utils that causes dev utils to break in IE11. strip-ansi package arrow function is causing this issue. See react-dev-utils patch PR that was accepted to fix this issue: https://github.com/facebook/create-react-app/pull/2692

jariz committed:
Merge branch 'master' into 365-package-manager-option

jaredpalmer reviewed (commented):
Nice
